File: internal/models/response.go
package models

import "keeper/internal/utils"

type SuccessResponse struct {
	Status  bool        `json:"status" swaggertype:"boolean"`
	Message string      `json:"message" swaggertype:"string"`
	Data    interface{} `json:"data,omitempty" swaggertype:"array,object"`
}

type PaginatedSuccessResponse struct {
	Status   bool           `json:"status" swaggertype:"boolean"`
	Message  string         `json:"message" swaggertype:"string"`
	Data     interface{}    `json:"data,omitempty" swaggertype:"array,object"`
	PageInfo utils.PageInfo `json:"page_info,omitempty"`
}

type ErrorResponse struct {
	Status bool   `json:"status" swaggertype:"boolean"`
	Error  string `json:"error" swaggertype:"string"`
}

File: client/src/common/types/response.ts
export type SuccessResponse<T> = {
  status: boolean;
  message: string;
  data: T;
};

export type PageInfo = {
  total_items: number;
  total_pages: number;
  has_next_page: boolean;
  current_page: number;
};

export type PaginatedSuccessResponse<T> = SuccessResponse<T> & {
  page_info: PageInfo;
};
